x-pack/plugins/metrics_entities/server/modules/host_entities/host_entities.json

```diff
@@ -0,0 +1,38 @@
+{
+  "id": "host_ent",
+  "description": "[host.name entities] grouped by @timestamp, host.name, os.name, and os.version, and aggregated on host.name",
+  "pivot": {
+    "group_by": {
+      "host.name": {
+        "terms": {
+          "field": "host.name"
+        }
+      },
+      "host.os.name": {
+        "terms": {
+          "field": "host.os.name",
+          "missing_bucket": true
+        }
+      },
+      "host.os.version": {
+        "terms": {
+          "field": "host.os.version",
+          "missing_bucket": true
+        }
+      },
+      "@timestamp": {
+        "date_histogram": {
```

Comment:
a date_histogram should always go 1st as it will allow the underlying composite aggregation to run faster, even better if you configured index sort on `@timestamp`.

Note: In future we will auto-optimize this, but for now we it depends on the configuration you put.

Reply:
Note: it is ALWAYS better to have `*histogram` group_by values near the top. Preferrably with your timestamp `date_histogram` at the top.

Reply:
Thanks for the tips. I will push my histograms to the top and have the topmost one be any configured with `@timetsamp`